src/ess/powder/correction.py

```diff
@@ -2,81 +2,113 @@
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
 """Correction algorithms for powder diffraction."""
 
-from typing import Any
+import enum
 
+import numpy as np
+import sciline
 import scipp as sc
-from scippneutron.conversion.graph import beamline, tof
 
 from ess.reduce.uncertainty import broadcast_uncertainties
 
 from ._util import event_or_outer_coord
-from .logging import get_logger
-from .smoothing import lowpass
 from .types import (
     AccumulatedProtonCharge,
-    FilteredData,
+    CaveMonitor,
+    DataWithScatteringCoordinates,
     FocussedDataDspacing,
     FocussedDataDspacingTwoTheta,
     IofDspacing,
     IofDspacingTwoTheta,
-    NormalizedByProtonCharge,
+    NormalizedRunData,
     RunType,
     SampleRun,
     UncertaintyBroadcastMode,
     VanadiumRun,
+    WavelengthMonitor,
 )
 
 
-def normalize_by_monitor(
-    data: sc.DataArray,
+def normalize_by_monitor_histogram(
+    detector: DataWithScatteringCoordinates[RunType],
     *,
-    monitor: sc.DataArray,
-    wavelength_edges: sc.Variable | None = None,
-    smooth_args: dict[str, Any] | None = None,
-) -> sc.DataArray:
+    monitor: WavelengthMonitor[RunType, CaveMonitor],
+    uncertainty_broadcast_mode: UncertaintyBroadcastMode,
+) -> NormalizedRunData[RunType]:
+    """Normalize detector data by a histogrammed monitor.
+
+    Parameters
+    ----------
+    detector:
+        Input event data in wavelength.
+    monitor:
+        A histogrammed monitor in wavelength.
+    uncertainty_broadcast_mode:
+        Choose how uncertainties of the monitor are broadcast to the sample data.
+
+    Returns
+    -------
+    :
+        `detector` normalized by a monitor.
     """
-    Normalize event data by a monitor.
+    norm = broadcast_uncertainties(
+        monitor, prototype=detector, mode=uncertainty_broadcast_mode
+    )
+    return detector.bins / sc.lookup(norm, dim="wavelength")
+
 
-    The input is converted to wavelength if it does not already contain wavelengths.
+def normalize_by_monitor_integrated(
+    detector: DataWithScatteringCoordinates[RunType],
+    *,
+    monitor: WavelengthMonitor[RunType, CaveMonitor],
+    uncertainty_broadcast_mode: UncertaintyBroadcastMode,
+) -> NormalizedRunData[RunType]:
+    """Normalize detector data by an integrated monitor.
+
+    The monitor is integrated according to
+
+    .. math::
+
+        M = \\sum_{i=0}^{N-1}\\, m_i (x_{i+1} - x_i) I(x_i, x_{i+1}),
+
+    where :math:`m_i` is the monitor intensity in bin :math:`i`,
+    :math:`x_i` is the lower bin edge of bin :math:`i`, and
+    :math:`I(x_i, x_{i+1})` selects bins that are within the range of the detector.
 
     Parameters
     ----------
-    data:
-        Input event data.
+    detector:
+        Input event data in wavelength.
     monitor:
-        A histogrammed monitor.
-    wavelength_edges:
-        If given, rebin the monitor with these edges.
-    smooth_args:
```

Comment:
So you removed the smoothing of the monitor. Was that intentional?

Reply:
Yes. This should be done as a separate pipeline step if desired.